IPG-Automotive-UK/react-ui
# TD-3067 Add LazyLoadImage RUI component

Closes [TD-3067](https://sce.myjetbrains.com/youtrack/issue/TD-3067/)

## Changes




- Wraps the `Infographic` component in a lazy load structure to only load the images we need. The base of the solution is the `IntersectionObserver`, in this case the bounding box is the reference
- Adds a `LazyLoadImage` component, which can be used instead of `<img>` tags to lazily load them. The base of the solution is the `IntersectionObserver`, observing the `<img>` tag in question.

This feature can only be tested in the browser. However, the affected components `Infographic`, `DetailCard` and `SummaryCard` had tests using the react testing library, so those tests would fail (same would apply to the new component `LazyLoadImage`). After talking about this issue with @dmbarry86, we came to the conclusion that creating new playwright tests would be a wasted effort, as there is an ongoing migration of the playwright tests to cypress. So I just left TODO comments in the component code, and commented out the tests, and renamed it to `.txt` files to not have failed tests (an empty test file means a fail to vitest). For this same reason, there is also no test created for the new `LazyLoadImage` component.

## Dependencies




## UI/UX




## Testing notes



You can open the storybook from https://666c1309face6e7c27f20e04-kwrjebdwct.chromatic.com/

**From my experience, storybook is can have problems with this lazy loading, e.g. in Firefox it is not a lazy load for me, while in Chrome works as a lazy loading in storybook. Therefore, to test I recommend using the deploy-preview I created in Virto, see the details in this PR: https://github.com/IPG-Automotive-UK/virto/pull/1029**

In storybook, the following places are relevant: `LazyLoadImage`, `DetailCard`, `SummaryCard` and `Infographic`

In those places:
- open the Network tab in the browser's dev tools, filter for images
- open the story
- remember which images loaded
- scroll down to the bottom
- as you scroll down, you should notice in the network tab that new images are being loaded
- in the card components and Infographic: make sure if a chip is mentioned in the story, that it is indeed there, inside the image

## Author checklist

Before I request a review:



- ~~[ ] I have tested the changes in Docker / a deploy-preview.~~ -> not relevant for RUI
- ~~[ ] I have included appropriate tests.~~ -> see the reason in the Description section
- ~~[ ] I have populated the deploy-preview with relevant test data.~~ -> not relevant for RUI
